syntax = "proto3";
option optimize_for = LITE_RUNTIME;

package video;

//房间表
message Room {
    string id           = 1; //房间id，唯一
    string author_id    = 2; //主播id
    string title        = 3; //标题
    string desc         = 4; //简介
    string cover_hash   = 5; //封面哈希
    int64  type         = 6; //房间类型 [1直播,2回放]
    int64  wish         = 7; //愿力 [打赏总值]
    int64  audience_size    = 8; //观众数量 [在线]
    int64  focus_size       = 9; //关注数量 [人气]
    string start_timestamp  = 10;//开始时间戳
    string end_timestamp    = 11;//结束时间戳
}

//供养表
message Fend {
    string id           = 1; //供养id，唯一
    string author_id    = 2; //主播id
    string room_id      = 3; //房间id
    string audience_id  = 4; //用户id
    int64  value        = 5; //供养值
}

//供养日志表
message FendLog {
    string id           = 1; //日志id，唯一
    string fend_id      = 2; //供养id
    string room_id      = 3; //房间id
    string audience_id  = 4; //用户id
    string goods_id     = 5; //产品id
    int64  goods_size   = 6; //产品数量
    int64  value        = 7; //供养值
    string timestamp    = 8; //时间戳
}

//关注表
message Focus {
    string room_id      = 1; //房间id, 前二者联合唯一
    string audience_id  = 2; //观众id
    string timestamp    = 3; //时间戳
}
